### PR TITLE
Text media sf

### PR DESCRIPTION
**Pull request by William-Blackie**
### Local development links for preview
Pattern-library component [here](http://0.0.0.0:3000/pattern-library/pattern/patterns/components/streamfields/text_and_media_block/text_and_media_block.html)
Pattern-library styleguide [here](http://0.0.0.0:3000/pattern-library/pattern/patterns/styleguide/text_and_medias.html) 

This PR includes: 

- Pattern-library components for the single streamfield and style guide to display the variations
- A override tag for `embed` tag so we can use it within the pattern library.
- Styles for the above component
- Small modification to the CTA component to pass styles through the template includes.

### CODE REVIEW

File: wagtailio/project_styleguide/templates/patterns/styleguide/text_and_medias.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<style>
```
Comment (zerolab) [inline wagtailio/project_styleguide/templates/patterns/styleguide/text_and_medias.html:1]: nitpick: media is a collective noun.

File: wagtailio/project_styleguide/templates/patterns/components/streamfields/quotes.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{% load wagtailcore_tags %}
```
Comment (zerolab) [inline wagtailio/project_styleguide/templates/patterns/components/streamfields/quotes.html:1]: this is.. out of place?

File: wagtailio/project_styleguide/templates/patterns/styleguide/text_and_medias.html (hunk 1 above)
Comment (William-Blackie) [reply wagtailio/project_styleguide/templates/patterns/styleguide/text_and_medias.html:1]: Praise: That's a good nitpick.

File: wagtailio/project_styleguide/templates/patterns/components/streamfields/quotes.html (hunk 2 above)
Comment (William-Blackie) [reply wagtailio/project_styleguide/templates/patterns/components/streamfields/quotes.html:1]: You're right, I didn't see this on the PR, I'm genuinely surprised how it even is here.

File: wagtailio/static/sass/components/_text-media.scss [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    $root: &;
+
+    &__content {
+        margin: auto;
+        grid-row: 2;
+        grid-column: 2 /span 2;
+        padding: 10px 10px 0 0;
+
+        @include media-query(medium) {
+            grid-column: 2/ span 2;
```
Comment (chris-lawton) [inline wagtailio/static/sass/components/_text-media.scss:11]: suggestion: remove this line as it's already set on line 7

File: wagtailio/static/sass/components/_text-media.scss [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            grid-row: 1;
+            padding: 0 40px 0 0;
+        }
+
+        @include media-query(large) {
+            grid-column: 2 / span 3;
+            grid-row: 1;
```
Comment (chris-lawton) [inline wagtailio/static/sass/components/_text-media.scss:18]: suggestion: remove this as it's the set on line 12

File: wagtailio/static/sass/components/_text-media.scss [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        height: 100%;
+        grid-column: 2 /span 2;
+        grid-row: 1;
+        margin: auto;
+
+        @include media-query(medium) {
+            grid-column: 4;
+            padding: 0 40px 0 0;
+        }
+        @include media-query(large) {
```
Comment (chris-lawton) [inline wagtailio/static/sass/components/_text-media.scss:40]:
```suggestion
        }
        
        @include media-query(large) {
```

**Comment by chris-lawton** [comment]
Looking good Will! I wonder if it would be worth stacking the image above the text for the medium breakpoint? The image gets squashed on this breakpoint atm:

Screenshot 2022-08-05 at 10 04 37

File: wagtailio/static/sass/components/_text-media.scss (hunk 3 above)
Comment (William-Blackie) [reply wagtailio/static/sass/components/_text-media.scss:11]: Good point!

File: wagtailio/static/sass/components/_text-media.scss (hunk 4 above)
Comment (William-Blackie) [reply wagtailio/static/sass/components/_text-media.scss:18]: A fair point!

**Comment by William-Blackie** [comment]
> Looking good Will! I wonder if it would be worth stacking the image above the text for the medium breakpoint? The image gets squashed on this breakpoint atm:
> 
> Screenshot 2022-08-05 at 10 04 37

Yep sure thing. Done.

**Pull request merged**